Security Incident Disclosure (#632)
Homebrew/brew.sh: _posts/2021-04-19-security-incident-disclosure.md

---
title: Security Incident Disclosure
author: Markus Reiter
---

> On 18th April 2021 a security researcher identified a vulnerability in our `review-cask-pr` GitHub action which is used to automatically review and approve version bump pull requests. This was reported via Hacker One and [a pull request demonstrating the vulnerability](https://github.com/Homebrew/homebrew-cask/pull/104191) was submitted in accordance with us. The [pull request was subsequently reverted](https://github.com/Homebrew/homebrew-cask/pull/104197) and the [`review-cask-pr` action disabled and removed](https://github.com/Homebrew/homebrew-cask/pull/104204) from all repositories that were using it.

Reviewer: Can we additionally note:
- it was merged by the automerge action and that it has also been disabled in favour of the GitHub feature in all homebrew-cask* repositories
- we've removed the ability for our bot(s) to commit to homebrew-cask* repositories
- we are moving forward with a more manual review process
- we are upskilling homebrew-core maintainers and documenting the process to make it easier to onboard homebrew-cask maintainers

> On 18th April 2021 a security researcher identified a vulnerability in our `review-cask-pr` GitHub action which is used to automatically review and approve version bump pull requests. This was reported via Hacker One and [a pull request demonstrating the vulnerability](https://github.com/Homebrew/homebrew-cask/pull/104191) was submitted in accordance with us. The [pull request was subsequently reverted](https://github.com/Homebrew/homebrew-cask/pull/104197) and the [`review-cask-pr` action disabled and removed](https://github.com/Homebrew/homebrew-cask/pull/104204) from all repositories that were using it.

Reviewer 2: ```suggestion
On 18th April 2021 a security researcher identified a vulnerability in our `review-cask-pr` GitHub action used to automatically review and approve version bump pull requests. This was reported via Hacker One and [a pull request demonstrating the vulnerability](https://github.com/Homebrew/homebrew-cask/pull/104191) was submitted in accordance with us. The [pull request was subsequently reverted](https://github.com/Homebrew/homebrew-cask/pull/104197) and the [`review-cask-pr` action disabled and removed](https://github.com/Homebrew/homebrew-cask/pull/104204) from all repositories that were using it.
```
I feel like "in accordance with us" can be reworded here, but I'm not sure about the best way to do so.

Reviewer 3: "with permission by us"?

> On 18th April 2021 a security researcher identified a vulnerability in our `review-cask-pr` GitHub action which is used to automatically review and approve version bump pull requests. This was reported via Hacker One and [a pull request demonstrating the vulnerability](https://github.com/Homebrew/homebrew-cask/pull/104191) was submitted in accordance with us. The [pull request was subsequently reverted](https://github.com/Homebrew/homebrew-cask/pull/104197) and the [`review-cask-pr` action disabled and removed](https://github.com/Homebrew/homebrew-cask/pull/104204) from all repositories that were using it.

Reviewer 4: ```suggestion
On 18th April 2021 a security researcher identified a vulnerability in our `review-cask-pr` GitHub action which is used to automatically review and approve version bump pull requests. This was reported via HackerOne and [a pull request demonstrating the vulnerability](https://github.com/Homebrew/homebrew-cask/pull/104191) was submitted in accordance with us. The [pull request was subsequently reverted](https://github.com/Homebrew/homebrew-cask/pull/104197) and the [`review-cask-pr` action disabled and removed](https://github.com/Homebrew/homebrew-cask/pull/104204) from all repositories that were using it.
```

I think it should be `HackerOne` as it's a company name: https://www.hackerone.com/company

> On 18th April 2021 a security researcher identified a vulnerability in our `review-cask-pr` GitHub action which is used to automatically review and approve version bump pull requests. This was reported via Hacker One and [a pull request demonstrating the vulnerability](https://github.com/Homebrew/homebrew-cask/pull/104191) was submitted in accordance with us. The [pull request was subsequently reverted](https://github.com/Homebrew/homebrew-cask/pull/104197) and the [`review-cask-pr` action disabled and removed](https://github.com/Homebrew/homebrew-cask/pull/104204) from all repositories that were using it.

Reviewer: ```suggestion
On 18th April 2021 [a security researcher identified](https://blog.ryotak.me/post/homebrew-security-incident-en/) a vulnerability in our `review-cask-pr` GitHub action which is used to automatically review and approve version bump pull requests. This was reported via Hacker One and [a pull request demonstrating the vulnerability](https://github.com/Homebrew/homebrew-cask/pull/104191) was submitted in accordance with us. The [pull request was subsequently reverted](https://github.com/Homebrew/homebrew-cask/pull/104197) and the [`review-cask-pr` action disabled and removed](https://github.com/Homebrew/homebrew-cask/pull/104204) from all repositories that were using it.
```

correct me if this is wrong @Ry0taK

Reviewer 4: That link is correct ;)

### What was impacted

> The discovered vulnerability would allow an attacker to inject arbitrary payloads into a cask and have it

Reviewer 4: ```suggestion
The discovered vulnerability would allow an attacker to inject arbitrary codes into a cask and have it
```
As I'm not familiar with English, I can be wrong.
However, I think `code` would be fit in this case.

> be merged automatically. This is due to a bug in the `git_diff` dependency of the `review-cask-pr` GitHub action which is used to parse a pull request's diff for inspection. Due to this bug, the parser can be spoofed into completely ignoring the offending lines, resulting in successfully approving a malicious pull request.

Reviewer 3: ```suggestion
be merged automatically. This is due to a flaw in the `git_diff` dependency of the `review-cask-pr` GitHub action which is used to parse a pull request's diff for inspection. Due to this flaw, the parser can be spoofed into completely ignoring the offending lines, resulting in successfully approving a malicious pull request.
```

Part of me feels that "flaw" is a better word to use instead of "bug" here. To me, bug implies something that we're fixing and I don't want to give the impression that we are going to fix the `review-cask-pr` action and continue using it.

> **A single cask was compromised with a dummy payload for the duration of [the demonstration pull request](https://github.com/Homebrew/homebrew-cask/pull/104191) until [its reversal](https://github.com/Homebrew/homebrew-cask/pull/104197). No action is required by users due to this incident.**

Reviewer 4: ```suggestion
**A single cask was compromised with a harmless code for the duration of [the demonstration pull request](https://github.com/Homebrew/homebrew-cask/pull/104191) until [its reversal](https://github.com/Homebrew/homebrew-cask/pull/104197). No action is required by users due to this incident.**
```
I feel `dummy payload` may be a little vague.
How about `harmless code`?

### What we're doing about it

> - The vulnerable `review-cask-pr` GitHub action has been disabled and removed from all repositories.

Reviewer 3: ```suggestion
- The vulnerable `review-cask-pr` GitHub action has been disabled and removed from all repositories.
- All pull requests will require a maintainer to manually approve and initiate a merge
```

It's implied but might be good to explicitly say this

We did, do and will continue to take the security of the project and our users very seriously. We try our best to behave as a for-profit company would do in terms of timely response to security issues but this is heavily limited by our lack of resources.

> In order to ensure and improve Homebrew's security, please consider contributing your code and code reviews to our GitHub projects, get in touch to volunteer to be on-call for security and/or system administration emergencies or [donate to the project](https://github.com/Homebrew/brew#donations).

Reviewer: ```suggestion
In order to ensure and improve Homebrew's security, please consider contributing your code and code reviews to our GitHub projects or [donate to the project](https://github.com/Homebrew/brew#donations).
```

Reviewer 5: I'd also prefer we not ask for donations here.

---
title: Security Incident Disclosure
author: Markus Reiter
---

On 18th April 2021 [a security researcher identified](https://blog.ryotak.me/post/homebrew-security-incident-en/) a vulnerability in our `review-cask-pr` GitHub action used to automatically review and approve version bump pull requests. This was reported via HackerOne and [a pull request demonstrating the vulnerability](https://github.com/Homebrew/homebrew-cask/pull/104191) was submitted with our permission. The [pull request was subsequently reverted](https://github.com/Homebrew/homebrew-cask/pull/104197) and the [`review-cask-pr` action disabled and removed](https://github.com/Homebrew/homebrew-cask/pull/104204) from all repositories that were using it.

### What was impacted

> The discovered vulnerability would allow an attacker to inject arbitrary payloads into a cask and have it be merged automatically. This is due to a flaw in the `git_diff` dependency of the `review-cask-pr` GitHub action which is used to parse a pull request's diff for inspection. Due to this flaw, the parser can be spoofed into completely ignoring the offending lines, resulting in successfully approving a malicious pull request.

Reviewer: ```suggestion
The discovered vulnerability would allow an attacker to inject arbitrary code into a cask and have it be merged automatically. This is due to a flaw in the `git_diff` dependency of the `review-cask-pr` GitHub action which is used to parse a pull request's diff for inspection. Due to this flaw, the parser can be spoofed into completely ignoring the offending lines, resulting in successfully approving a malicious pull request.
```

> **A single cask was compromised with a dummy payload for the duration of [the demonstration pull request](https://github.com/Homebrew/homebrew-cask/pull/104191) until [its reversal](https://github.com/Homebrew/homebrew-cask/pull/104197). No action is required by users due to this incident.**

Reviewer: ```suggestion
**A single cask was compromised with a harmless change for the duration of [the demonstration pull request](https://github.com/Homebrew/homebrew-cask/pull/104191) until [its reversal](https://github.com/Homebrew/homebrew-cask/pull/104197). No action is required by users due to this incident.**
```
